Source/WebCore/html/CheckboxInputType.cpp

```diff
@@ -159,45 +170,44 @@ void CheckboxInputType::handleTouchEvent(TouchEvent& event)
     const AtomString& eventType = event.type();
     auto& eventNames = WebCore::eventNames();
     if (eventType == eventNames.touchstartEvent) {
-        RefPtr<TouchList> targetTouches = event.targetTouches();
-        if (!targetTouches)
-            return;
-        if (targetTouches->length() != 1)
+        RefPtr targetTouches = event.targetTouches();
+        if (!targetTouches || targetTouches->length() != 1)
             return;
-        RefPtr<Touch> touch = targetTouches->item(0);
-
-        startSwitchPointerTracking({ touch->pageX(), touch->pageY() }, touch->identifier());
-        performSwitchAnimation(SwitchAnimationType::Pressed);
+        RefPtr touch = targetTouches->item(0);
+
+        m_switchPointerTrackingTouchIdentifier = touch->identifier();
+        if (!m_switchHeldTimer) {
+            m_switchHeldTimer = makeUnique<Timer>([protectedThis = Ref { *this }, touch] {
+                if (!protectedThis->isSwitch() || !protectedThis->element() || !protectedThis->element()->renderer())
+                    return;
+                protectedThis->startSwitchPointerTracking({ touch->pageX(), touch->pageY() });
+                protectedThis->setIsSwitchHeld(true);
+            });
+        }
+        m_switchHeldTimer->startOneShot(200_ms);
```

Comment:
`200_ms` should probably be in a `static constexpr` variable near the top of this file.

Similar to `static constexpr unsigned maxSelectItems = 100000;` in `HTMLSelectElement.cpp`.